accumulo/src/main/java/com/yahoo/ycsb/db/accumulo/AccumuloClient.java

```diff
@@ -100,62 +104,72 @@ public void init() throws DBException {
   @Override
   public void cleanup() throws DBException {
     try {
-      if (bw != null) {
-        bw.close();
+      Iterator<BatchWriter> iterator = writers.values().iterator();
+      while (iterator.hasNext()) {
+        BatchWriter writer = iterator.next();
+        writer.close();
+        iterator.remove();
       }
     } catch (MutationsRejectedException e) {
       throw new DBException(e);
     }
   }
 
   /**
-   * Commonly repeated functionality: Before doing any operation, make sure
-   * we're working on the correct table. If not, open the correct one.
-   * 
-   * @param t
-   *          The table to open.
-   */
-  public void checkTable(String t) throws TableNotFoundException {
-    if (!table.equals(t)) {
-      getTable(t);
-    }
-  }
-
-  /**
    * Called when the user specifies a table that isn't the same as the existing
    * table. Connect to it and if necessary, close our current connection.
    * 
    * @param t
    *          The table to open.
    */
-  public void getTable(String t) throws TableNotFoundException {
-    if (bw != null) { // Close the existing writer if necessary.
-      try {
-        bw.close();
-      } catch (MutationsRejectedException e) {
-        // Couldn't spit out the mutations we wanted.
-        // Ignore this for now.
-        System.err.println("MutationsRejectedException: " + e.getMessage());
+  public BatchWriter getWriter(String t) throws TableNotFoundException {
```

Comment:
Ahh, good catch. This was originally because of the `table` member (checkstyle rule about a local variable masking a class member). Will fix.